experiments/main_solver/run.py

```
@@ -0,0 +1,357 @@
+def _run_command(
+    command: list[str],
+    *,
+    cwd: Path,
+    stdout_path: Path,
+    stderr_path: Path,
+) -> dict[str, Any]:
+    stdout_path.parent.mkdir(parents=True, exist_ok=True)
+    start = time.monotonic()
+    with stdout_path.open("w", encoding="utf-8") as stdout_obj:
+        with stderr_path.open("w", encoding="utf-8") as stderr_obj:
+            completed = subprocess.run(
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)